front/src/components/header/header.tsx

```diff
@@ -0,0 +1,23 @@
+import styles from "./header.module.css";
+
+type Props = {
+    characterName: string;
+};
+
+const Header = ({ characterName }: Props) => {
```

Comment:
A lot of people tend to delcare functions this way without properly understanding why people do it. When we're dealing with context and global context, this method naturally binds `this` to the current context, not the global one. However, in modern Javascript/React, we're almost never referencing `this`, especially because we're using Functional Components and not the older Class Components.

Using `function` to declare functions gives better runtime debugging, and it's just easier to read, given that it's more explicit.

```suggestion
function Header({ characterName }: Props) {
```